integration/hibernate-4/src/main/java/com/blazebit/persistence/impl/hibernate/Hibernate4Integrator.java

```diff
@@ -36,7 +37,8 @@ public void integrate(Configuration configuration, SessionFactoryImplementor ses
 				// TODO: check that no collections are mapped
 			}
 		}
-		CommonHibernateUtils.setCollectionPersisterClass(configuration.getClassMappings(), CustomOneToManyPersister.class);
+
+		serviceRegistry.locateServiceBinding(PersisterClassResolver.class).setService(new CustomPersisterClassResolver());
```

Comment:
Ah yes, printing a warning using this mechanism is not easily done, because Hibernate only invokes the `PersisterClassResolver` if a custom persister is not defined. On the other hand, it seems that using the `PersisterClassResolver` is more in line with the Hibernate SPI. Furthermore it seems very rare to me that one would use a custom persister, with CTE queries, and the `FetchMode.SUBSELECT`.

So adding some notes on how the fetch support works seems sufficient to me. I will amend that to my PR.